wolfcrypt/src/evp.c

```diff
@@ -6328,6 +6328,32 @@ const WOLFSSL_EVP_MD* wolfSSL_EVP_get_digestbynid(int id)
     return NULL;
 }
 
+/* Returns the NID associated with the WOLFSSL_EVP_MD argument.
+ *
+ * md - pointer to WOLFSSL_EVP_MD for which to return NID value
+ *
+ * Returns NID on success, BAD_FUNC_ARG on bad WOLFSSL_EVP_MD value.
+ */
+int wolfSSL_EVP_MD_nid(const WOLFSSL_EVP_MD* md)
+{
+    const struct s_ent *ent;
+
+    WOLFSSL_ENTER("wolfSSL_EVP_MD_nid");
+
+    if (md == NULL) {
+        WOLFSSL_MSG("No md type arg");
+        return BAD_FUNC_ARG;
+    }
+
+    for (ent = md_tbl; ent->name != NULL; ent++) {
+        if (XSTRNCMP((const char*)md, ent->name, XSTRLEN(ent->name)+1) == 0) {
+            return ent->nid;
+        }
+    }
+
+    return BAD_FUNC_ARG;
```

Comment:
> EVP_MD_get_type(), EVP_MD_get_pkey_type()
Returns the NID of the corresponding OBJECT IDENTIFIER or NID_undef if none exists.

Looks like this should return `NID_undef` on failure, rather than `BAD_FUNC_ARG`. Also, it looks like OpenSSL maps this to `EVP_MD_type`, which we already support, so we could just `#define EVP_MD_nid wolfSSL_EVP_MD_type`.

Reply:
YES. Please `#define EVP_MD_nid wolfSSL_EVP_MD_type` and change the return in `wolfSSL_EVP_MD_type` to `NID_undef`.

Reply:
Excellent feedback, adjusted to match recommendations